core/lib/vm_executor/src/batch/factory.rs

```diff
@@ -356,13 +360,18 @@ impl<S: ReadStorage + 'static, Tr: BatchTracer> CommandReceiver<S, Tr> {
             }
         }
 
+        let mut has_snapshot_before_tx = false;
```

Comment:
oh uh ... Is it hard to have maybe two snapshot Queues - one before tx and one before block? If not, we'd need to do it like this but it's so error prone..